MapboxCoreNavigation/String.swift

```diff
@@ -50,4 +50,37 @@ extension String {
     var containsDecimalDigit: Bool {
         return self.rangeOfCharacter(from: CharacterSet.decimalDigits) != nil
     }
+    
+    // Adapted from https://github.com/raywenderlich/swift-algorithm-club/blob/master/Minimum%20Edit%20Distance/MinimumEditDistance.playground/Contents.swift
```

Comment:
This code is substantially similar to that playground code. We’ll need to include [this license](https://github.com/raywenderlich/swift-algorithm-club/blob/master/LICENSE.txt) either right here in this file or in LICENSE.md. That isn’t a problem at all, but here are two alternatives anyways:

* Write the code from scratch according to the well-documented [Wagner–Fischer algorithm](https://en.wikipedia.org/wiki/Wagner–Fischer_algorithm).
* Depend on [this library](https://cocoapods.org/pods/MDCDamerauLevenshtein) instead, because it provides a [Damerau–Levenshtein distance](https://en.wikipedia.org/wiki/Damerau–Levenshtein_distance) function. Unlike a basic Levenshtein distance, Damerau–Levenshtein treats a transposition (like swapping two words) as a single edit instead of a deletion and addition. We’ve had good results with this approach in a previous application.

Reply:
I'd really like to steer clear of adding another dependency. I'm 👍  on adding this to a note LICENSE.md.

Reply:
FYI, Damerau–Levenshtein is quite concise, so if adding a dependency is a problem, it shouldn’t be too difficult to port [this code](https://github.com/modocache/MDCDamerauLevenshtein/blob/91eae31ef9bff5396371471518e554c6020a6913/MDCDamerauLevenshtein/Algorithms/MDCDamerauLevenshteinDistance.m#L16) to Swift and add a similar MIT license block to the codebase.